wiseconnect/CMakeLists.txt

```diff
@@ -61,44 +97,26 @@ zephyr_include_directories(
     components/device/silabs/si91x/mcu/core/chip/config
     components/protocol/wifi/inc
     components/service/network_manager/inc
-    resources/defaults
 )
-
-zephyr_compile_definitions(
-    SLI_SI91X_OFFLOAD_NETWORK_STACK
-    SLI_SI91X_SOCKETS
+zephyr_compile_definitions_ifdef(CONFIG_SIWX917_NWP_NETWORK_STACK
```

Comment:
Do we want to name this option `CONFIG_SIWX917_NWP_NETWORK_STACK` or `CONFIG_WISECONNECT_NETWORK_STACK`?

Reply:
Currently these are defined in `soc/silabs/silabs_siwx917/siwg917/Kconfig` in the main tree. If we change the name to `WISECONNECT_*` I think a more appropriate place would be `modules/Kconfig.silabs` (also in the main tree).

Reply:
Yes, we can consider this option is more a "mode of operation" of WiseConnect rather than just a support for a driver. I feel it make sense to hide it in `modules/Kconfig.silabs` (note this file is currently named `modules/Kconfig` in our downstream)

Reply:
> Yes, we can consider this option is more a "mode of operation" of WiseConnect rather than just a support for a driver. I feel it make sense to hide it in `modules/Kconfig.silabs`

Sure, makes sense. Unrelated to the current case, but I have a feeling we may want to move to using `kconfig-ext` in `module.yml` and then use `modules/hal_silabs/Kconfig` in the main tree. That seems to be the more "modern" approach.

> note this file is currently named modules/Kconfig in our downstream

Hmm? I can't find a modules directory at all in zephyr-silabs.

Reply:
> Is it ok to do just the rename here and do the relocation in another PR?

So you agree this can be seen as a more WiseConnect specific thing rather than a 917 specific thing? I'm not opposed to just doing the rename at first, though in both cases you will anyway need to update your zephyr-silabs PR, i.e. it's not really that much additional work to create a `modules/Kconfig.silabs` that will eventually get merged with the corresponding file in the main tree.

Btw, I realized that we have `config HAS_SILABS_WISECONNECT` in the zephyr-silabs root Kconfig, which should really also be in the module Kconfig file.

Reply:
>> note this file is currently named modules/Kconfig in our downstream
[...]
> Btw, I realized that we have config HAS_SILABS_WISECONNECT in the zephyr-silabs root  Kconfig, which should really also be in the module Kconfig file.

Yes, it was what I wanted to say.

Reply:
Yes, completely agree. 
You're right, I need to update both PR so then I'll do these Kconfig modifications as well.

Reply:
@silabs-TiborL I think there was some misunderstanding. The place I was suggesting to put this was in the `zephyr-silabs` tree and not the `hal_silabs` tree. The goal is also to remove all CMake and Kconfig files completely from `hal_silabs` and have those instead in the main tree (or `zephyr-silabs` in the case of pre-upstream 917 work).

For now I'd create a `modules/Kconfig.silabs` file in the `zephyr-silabs` tree and put all of this there. Please also see my comment in the #zephyr-development Slack channel how this should change in the long-term (however for now `modules/Kconfig.silabs` is the way to go, since that mirrors what's in the upstream main tree).

Reply:
I see, sorry about this, I've only started working with Zephyr recently andI'm not 100% sure that I got your intention correctly. Thus I'd go for the easy path and just rename CONFIG_SIWX917_NWP_NETWORK_STACK to CONFIG_WISECONNECT_NETWORK_STACK for this PR.

Reply:
Sure. Let's go with that then in the short-term.